trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/IcebergDataTypeTransformer.java

```diff
@@ -59,14 +59,50 @@ public io.trino.spi.type.Type getTrinoType(Type type) {
     if (Name.FIXED == type.name()) {
       return VarbinaryType.VARBINARY;
     } else if (Name.TIME == type.name()) {
+      Types.TimeType timeType = (Types.TimeType) type;
+      if (timeType.hasPrecisionSet()) {
+        int precision = timeType.precision();
+        if (precision == TRINO_SECONDS_PRECISION) {
+          return TimeType.TIME_SECONDS;
+        } else if (precision <= TRINO_MILLIS_PRECISION) {
+          return TimeType.TIME_MILLIS;
+        } else if (precision <= TRINO_MICROS_PRECISION) {
+          return TimeType.TIME_MICROS;
```

Comment:
why don't you use `TimeType.createTimeType(precision)` to use the corresponding precision?

Reply:
done